File: SproInstall/SocketPro/tutorial/CSharp/MySecure/MySecureClient/frmMyClient.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using SocketProAdapter;
using SocketProAdapter.ClientSide;
using USOCKETLib;

namespace MySecureClient
{
    public partial class frmMySecure : Form
    {
        private CClientSocket m_ClientSocket = new CClientSocket();
        private CMySecure m_AsynHandler = new CMySecure();

        public frmMySecure()
        {
            InitializeComponent();
        }

        private void btnConnect_Click(object sender, EventArgs e)
        {
            m_ClientSocket.Connect(txtAddress.Text, int.Parse(txtPort.Text));
        }

        private void btnDisconnect_Click(object sender, EventArgs e)
        {
            m_ClientSocket.Disconnect();
        }

        private void OnClosed(int nSocketHandle, int nError)
        {
            btnToDB.Enabled = false;
            btnExecuteSQL.Enabled = false;
        }

        private void OnConnected(int nSocketHandle, int nError)
        {
            if (nError == 0)
            {
                btnToDB.Enabled = true;

                m_ClientSocket.EncryptionMethod = tagEncryptionMethod.BlowFish;

                //set user id and password
                m_ClientSocket.SetUID(txtUserID.Text);
                m_ClientSocket.SetPassword(txtPassword.Text);

                //switch for the service identified by sidSOneSvs
                m_ClientSocket.SwitchTo(m_AsynHandler);
            }
            else
            {
                MessageBox.Show(m_ClientSocket.GetErrorMsg());
            }
        }
       
        private void frmMySecure_Load(object sender, EventArgs e)
        {
            m_ClientSocket.m_OnSocketClosed += new DOnSocketClosed(OnClosed);
            m_ClientSocket.m_OnSocketConnected += new DOnSocketConnected(OnConnected);
            m_AsynHandler.Attach(m_ClientSocket);
        }

        private void btnToDB_Click(object sender, EventArgs e)
        {
            USocketClass usc = m_ClientSocket.GetUSocket();

            //we like to encrypt data before sending the request to server
            usc.EncryptionMethod = (short)USOCKETLib.tagEncryptionMethod.BlowFish;

            if (m_AsynHandler.Open(txtUserID.Text, txtPassword.Text) == null)
                MessageBox.Show(m_AsynHandler.m_strErrorMessage);
            else
                btnExecuteSQL.Enabled = true;
        }

        private void btnExecuteSQL_Click(object sender, EventArgs e)
        {
            USocketClass usc = m_ClientSocket.GetUSocket();

            //we do not want to encrypt data for the below requests
            usc.EncryptionMethod = (short)USOCKETLib.tagEncryptionMethod.NoEncryption;
            m_ClientSocket.BeginBatching();
            m_AsynHandler.BeginTransAsyn();
            m_AsynHandler.ExecuteNoQueryAsyn(txtSQL.Text);
            m_AsynHandler.CommitAsyn(true);
            m_ClientSocket.Commit(true);
            m_ClientSocket.WaitAll();
        }
    }
}

File: SproInstall/SocketPro/tutorial/VBNet/MySecure/MySecureClient/frmMyClient.vb
Imports System.ComponentModel
Imports System.Text
Imports SocketProAdapter
Imports SocketProAdapter.ClientSide
Imports USOCKETLib

Namespace MySecureClient
	Partial Public Class frmMySecure
		Inherits Form
		Private m_ClientSocket As New CClientSocket()
        Private m_AsynHandler As New CMySecure()

		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub btnConnect_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnConnect.Click
			m_ClientSocket.Connect(txtAddress.Text, Integer.Parse(txtPort.Text))
		End Sub

		Private Sub btnDisconnect_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnDisconnect.Click
			m_ClientSocket.Disconnect()
		End Sub

		Private Overloads Sub OnClosed(ByVal nSocketHandle As Integer, ByVal nError As Integer)
			btnToDB.Enabled = False
			btnExecuteSQL.Enabled = False
		End Sub

		Private Sub OnConnected(ByVal nSocketHandle As Integer, ByVal nError As Integer)
			If nError = 0 Then
				btnToDB.Enabled = True

                m_ClientSocket.EncryptionMethod = USOCKETLib.tagEncryptionMethod.BlowFish

				'set user id and password
				m_ClientSocket.SetUID(txtUserID.Text)
				m_ClientSocket.SetPassword(txtPassword.Text)

				'switch for the service identified by sidSOneSvs
				m_ClientSocket.SwitchTo(m_AsynHandler)
			Else
				MessageBox.Show(m_ClientSocket.GetErrorMsg())
			End If
		End Sub

		Private Sub frmMySecure_Load(ByVal sender As Object, ByVal e As EventArgs) Handles MyBase.Load
            m_ClientSocket.m_OnSocketClosed = AddressOf OnClosed
            m_ClientSocket.m_OnSocketConnected = AddressOf OnConnected
			m_AsynHandler.Attach(m_ClientSocket)
		End Sub

		Private Sub btnToDB_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnToDB.Click
			Dim usc As USocketClass = m_ClientSocket.GetUSocket()

			'we like to encrypt data before sending the request to server
			usc.EncryptionMethod = CShort(Fix(USOCKETLib.tagEncryptionMethod.BlowFish))

			If m_AsynHandler.Open(txtUserID.Text, txtPassword.Text) Is Nothing Then
				MessageBox.Show(m_AsynHandler.m_strErrorMessage)
			Else
				btnExecuteSQL.Enabled = True
			End If
		End Sub

		Private Sub btnExecuteSQL_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnExecuteSQL.Click
			Dim usc As USocketClass = m_ClientSocket.GetUSocket()

			'we do not want to encrypt data for the below requests
			usc.EncryptionMethod = CShort(Fix(USOCKETLib.tagEncryptionMethod.NoEncryption))
			m_ClientSocket.BeginBatching()
			m_AsynHandler.BeginTransAsyn()
			m_AsynHandler.ExecuteNoQueryAsyn(txtSQL.Text)
			m_AsynHandler.CommitAsyn(True)
			m_ClientSocket.Commit(True)
			m_ClientSocket.WaitAll()
		End Sub
	End Class
End Namespace